pgmax/fg/graph.py

```diff
@@ -432,12 +526,23 @@ def __getitem__(self, name: Any) -> np.ndarray:
 
         if name in self.fg_state.named_factor_groups:
             factor_group = self.fg_state.named_factor_groups[name]
+            if not isinstance(
```

Comment:
This is again a factor type dependent special processing. This means when adding new factors we potentially need to touch this file, which we want to avoid.

I don't see any harm in having placeholder values for log potentials in logical factors. Currently it's None. But we can use any placeholder value. Once we have that we can actually all the factors the same way. In particular we can keep the previous implementation of register factor group, and no special processing is needed here.